### PR TITLE
Replace Wallet onboarding conversion P3A question with a more precise question

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#26011

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

0. Start with fresh profile, ensure metric does not exist on local state page.
1. Open Wallet onboarding page and exit. Ensure 0 is recorded for the metric
2. Create a new wallet, take note of the recovery phrase. Ensure 1 is recorded for the metric.
3. Add/import a secondary wallet. Ensure the metric value is the same.
4. Lock the wallet and restore using the recovery phrase. Ensure the metric value is the same.
5. Reset the wallet, restore the wallet using the recovery phrase. Ensure 0 is recorded before recovery and 2 is recorded after recovery.

### CODE REVIEW

File: components/brave_wallet_ui/page/async/wallet_page_async_handler.ts [hunk 1]
```diff
@@ -102,6 +103,8 @@ handler.on(WalletPageActions.restoreWallet.getType(), async (store: Store, paylo
   store.dispatch(WalletPageActions.setShowIsRestoring(false))
   if (payload?.completeWalletSetup) {
     store.dispatch(WalletPageActions.walletSetupComplete(payload.completeWalletSetup))
+  } else {
+    store.dispatch(WalletPageActions.reportOnboardingAction(OnboardingAction.RESTORED_WALLET))
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:107]: No need to dispatch an extra action here. You can call `getWalletPageApiProxy().braveWalletP3A.reportOnboardingAction` directly here

File: components/brave_wallet_ui/page/async/wallet_page_async_handler.ts [hunk 2]
```diff
@@ -299,9 +302,8 @@ handler.on(WalletPageActions.getNFTMetadata.getType(), async (store, payload: Br
   store.dispatch(WalletPageActions.setIsFetchingNFTMetadata(false))
 })
 
-handler.on(WalletPageActions.onOnboardingShown.getType(), async (store: Store) => {
-  const braveWalletService = getWalletPageApiProxy().braveWalletService
-  await braveWalletService.onOnboardingShown()
+handler.on(WalletPageActions.reportOnboardingAction.getType(), async (store: Store, payload: OnboardingAction) => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/async/wallet_page_async_handler.ts:305]: We don't need this action/reducer anymore. The Api proxy can now be accessed from React components directly using the `useApiProxy` hook.

File: components/brave_wallet_ui/page/screens/onboarding/create-password/onboarding-create-password.tsx [hunk 3]
```diff
@@ -59,6 +59,7 @@ export const OnboardingCreatePassword = () => {
   // effects
   React.useEffect(() => {
     if (isWalletCreated) {
+      dispatch(WalletPageActions.reportOnboardingAction(OnboardingAction.CREATED_WALLET))
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/onboarding/create-password/onboarding-create-password.tsx:62]: You can access the braveWalletP3A object without using redux actions like this:

```
// custom hooks
  const { braveWalletP3A } = useApiProxy()
  
  // effects
  React.useEffect(() => {
    if (isWalletCreated) {
      braveWalletP3A.reportOnboardingAction(OnboardingAction.CREATED_WALLET))
      history.push(WalletRoutes.OnboardingExplainRecoveryPhrase)
    }
  }, [isWalletCreated, braveWalletP3A])
```

File: components/brave_wallet_ui/page/screens/onboarding/welcome/onboarding-welcome.tsx [hunk 4]
```diff
@@ -61,7 +61,7 @@ export const OnboardingWelcome = () => {
     if (!setupStillInProgress) {
       dispatch(WalletPageActions.walletSetupComplete(false))
     }
-    dispatch(WalletPageActions.onOnboardingShown())
+    dispatch(WalletPageActions.reportOnboardingAction(OnboardingAction.SHOWN))
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/onboarding/welcome/onboarding-welcome.tsx:64]: Ditto: use the `useApiProxy` hook instead of dispatching

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Review by josheleonard — APPROVED** [review]
Desktop Frontend ++

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java [hunk 5]
```diff
@@ -243,15 +244,15 @@ private void setNavigationFragments(int type) {
...
+            mBraveWalletP3A.reportOnboardingAction(OnboardingAction.SHOWN);
         } else if (type == UNLOCK_WALLET_ACTION) {
             UnlockWalletFragment unlockWalletFragment = new UnlockWalletFragment();
             unlockWalletFragment.setOnNextPageListener(this);
             navigationItems.add(new NavigationItem(
                     getResources().getString(R.string.unlock_wallet_title), unlockWalletFragment));
         } else if (type == RESTORE_WALLET_ACTION) {
             mShowBiometricPrompt = false;
-            RestoreWalletFragment restoreWalletFragment = new RestoreWalletFragment();
+            RestoreWalletFragment restoreWalletFragment = new RestoreWalletFragment(false);
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java:255]: we should not pass arguments to constructor. instead we should pass the argument to fragment.

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java [hunk 6]
```diff
@@ -280,12 +281,13 @@ private void addRemoveSecureFlag(boolean add) {
...
     }
 
-    private void replaceNavigationFragments(int type, boolean doNavigate) {
+    private void replaceNavigationFragments(int type, boolean doNavigate, boolean isOnboarding) {
         mShowBiometricPrompt = true;
         if (cryptoWalletOnboardingViewPager != null && cryptoWalletOnboardingPagerAdapter != null) {
             if (type == RESTORE_WALLET_ACTION) {
                 mShowBiometricPrompt = false;
-                RestoreWalletFragment restoreWalletFragment = new RestoreWalletFragment();
+                RestoreWalletFragment restoreWalletFragment =
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java:289]: we should not pass arguments to constructor. instead we should pass the argument to fragment.

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java [hunk 7]
```diff
@@ -37,6 +39,12 @@ public class RestoreWalletFragment extends CryptoOnboardingFragment {
     private CheckBox showRecoveryPhraseCheckbox;
     private CheckBox restoreLegacyWalletCheckbox;
     private boolean isLegacyWalletRestoreEnable;
+    private boolean isOnboarding;
+
+    public RestoreWalletFragment(boolean isOnboarding) {
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java:44]: we shouldn't create a constructor for argument, instead we should pass arguments. Configuration change would use the default constructor and you may loose the value for argument.

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java (hunk 5 above)
Comment (DJAndries) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java:255]: can you be more specific

**Review by iefremov — APPROVED** [review]
src/components lgtm

**Pull request merged**